package.json

```diff
@@ -38,6 +38,7 @@
     "fix": "eslint . --fix",
     "lint": "eslint .",
     "make-checkpoints": "node -r sucrase/register scripts/makeCheckpoints.ts",
+    "make-cosmos-protos": "for file in ./scripts/cosmos-ts-protos/*.sh; do [ -f \"$file\" ] && chmod +x \"$file\" && \"./$file\"; done",
```

Comment:
Did ChatGPT write this stuff? It seems weirdly over-complicated. We just want to download a zip file and run `protoc` on its contents.

I suggest creating a `/scripts/makeCosmosProtos.sh` that does something like:

```shell
set -e

makeCosmosProtos () {
  local name="$1"
  local url="$2"
  local zipPath = "./tmp/$name.zip"
  
  # Download and unpack the zip: 
  wget "$url" -o $zipPath;
  unzip $zipPath;

  # Create the proto files:  
  for FILE in $PROTO_FILES; do
    echo "Creating typescript file from" $FILE
    protoc \
      --plugin="$(yarn bin protoc-gen-ts_proto 2> >(grep -v warning 1>&2))" \
      --ts_proto_out="$OUT_DIR" \
      --proto_path="$COSMOS_PROTO_DIR" \
      --proto_path="$THIRD_PARTY_PROTO_DIR" \
      --ts_proto_opt="esModuleInterop=true,forceLong=long,useOptionals=true" \
      $FILE
  done
}

makeCosmosProtos(thorchain, "https://gitlab.com/thorchain/thornode/-/archive/mainnet/thornode-mainnet.zip")
```

It's a lot less ceremony, and since we've put the unpack-and-generate step into a function, we can just add another call at the bottom for each chain we add. This is a lot safer than iterating over files in a directory and setting +x on them, that's for sure.

Even better, what about converting this from Shell to TypeScript? We already do *extremely* similar stuff in the react-native-mymonero-core repo, where we download zip files, unpack them, and process the contents with shell commands. Using that as a template would be even better still, since we know how to write and code-review TypeScript all day long. We don't know shell as well.